src/blocksvd.jl

```diff
@@ -0,0 +1,35 @@
+#=
+SVD on blockmatrices:
+Interpret the matrix as a linear map acting on vector spaces with a direct sum structure, which is reflected in the structure of U and V.
+In the generic case, the SVD does not preserve this structure, and can mix up the blocks, so S becomes a single block.
+(Implementation-wise, also most efficiently done by first mapping to a `BlockedArray`)
+In the case of `BlockDiagonal` however, the structure is preserved and carried over to the structure of `S`.
+=#
+
+LinearAlgebra.eigencopy_oftype(A::AbstractBlockMatrix, S) = BlockedArray(Array{S}(A), blocksizes(A, 1), blocksizes(A, 2))
```

Comment:
This would be simpler and more generic:
```suggestion
LinearAlgebra.eigencopy_oftype(A::AbstractBlockMatrix, S) = BlockedArray{S}(A)
```